lntest/miner/miner.go

```diff
@@ -292,19 +291,23 @@ func (h *HarnessMiner) AssertTxInMempool(txid *chainhash.Hash) *wire.MsgTx {
 			return fmt.Errorf("empty mempool")
 		}
 
-		for _, memTx := range mempool {
-			// Check the values are equal.
-			if *memTx == *txid {
-				return nil
-			}
+		isEqual := func(memTx *chainhash.Hash) bool {
+			return *memTx == *txid
 		}
+		result := fn.Find(isEqual, mempool)
 
-		return fmt.Errorf("txid %v not found in mempool: %v", txid,
-			mempool)
+		if result.IsNone() {
+			return fmt.Errorf("txid %v not found in "+
+				"mempool: %v", txid, mempool)
+		}
+
+		return nil
 	}, wait.MinerMempoolTimeout)
 
 	require.NoError(h, err, "timeout checking mempool")
 
+	msgTx := h.GetRawTransaction(txid).MsgTx()
+
 	return msgTx
```

Comment:
Consolidate this:

```golang
return h.GetRawTransaction(txid).MsgTx()
```

Reply:
totally, will push the change until I receive the final comments from yong.